### PR TITLE
Use game state structure for last map window loco globals issue #1132

### PR DESCRIPTION
**Pull request by memellis**
Happy to update as required from review.
I'm not sure if I got the structure of the changes right.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Looks like you didn't branch off from an upstream master before making your changes. Could you rebase and force-push your branch, such that only the final commit remains?

**Comment by memellis** [comment]
Yes I will  rebase and force-push your branch, such that only the final commit remains.
Sorry about that. 
It seems trying to reuse a fork is not a good idea :(

**Comment by AaronVanGeffen** [comment]
Reusing a fork is fine, so long as you make sure `master` isn't tracking your fork, but OpenLoco's master ('upstream'). Perhaps this article from our friends over at OpenRCT2 will help: [Rebase and Sync fork with OpenRCT2](https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2)

**Comment by memellis** [comment]
Ah thanks I'll take a look.

**Comment by memellis** [comment]
Frustratingly I can't seem to rebase to only have my required commits in this pull request.
So I propose closing the pull request, deleting my fork, creating a new fork with a pull request with only the commits I want.

**Comment by AaronVanGeffen** [comment]
You seem to have merged master in here, rather than performing a rebase. What was your logic for doing so?

I would encourage you not to start over with a new fork. Learning how to perform a rebase properly will help you in the long run.

**Comment by memellis** [comment]
Ok I will persist so I can learn how to rebase properly.
I've been trying to rebase as per the OpenRCT2 instructions.
It wasn't my intent to merge master but have done so in error.
I've thought I had git commands that showed me what is just on my branch.
I'll do some more research to remove the commits I don't want.

File: src/OpenLoco/src/LastMapWindowAttributes.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#include <cstdint>
+#include "Window.h"
+
+namespace OpenLoco::LastMapWindow
+{
+#pragma pack(push, 1)
+    struct LastMapWindowAttributes
+    {
+        Ui::WindowFlags lastMapWindowFlags; // 0x00526284
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/LastMapWindowAttributes.h:11]: no need to repeat lasMapWindow in this structure you would already know its the last map window by the struct.

File: src/OpenLoco/src/LastMapWindowAttributes.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+#include "Window.h"
+
+namespace OpenLoco::LastMapWindow
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/LastMapWindowAttributes.h:6]: It is perhaps a bit overkill to put this in a namespace on its own. The map window is part of the UI. I would suggest putting it in the UI namespace.

File: src/OpenLoco/src/Windows/MapWindow.cpp [hunk 3]
```diff
@@ -112,6 +109,8 @@ namespace OpenLoco::Ui::Windows::MapWindow
 
     static WindowEventList events;
 
+    auto& lastMapWindowAttributes = LastMapWindow::getLastMapWindowAttributes();
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/MapWindow.cpp:112]: would prefer you don't get it like this with a global. Just get it where its used.

File: src/OpenLoco/CMakeLists.txt [hunk 4]
```diff
@@ -108,6 +108,7 @@ set(OLOCO_SOURCES
     "${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Hooks.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/Intro.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/LastGameOptionManager.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/LastMapWindowAttributes.cpp"
```
Comment (duncanspumpkin) [inline src/OpenLoco/CMakeLists.txt:111]: After changing this to the Ui namespace you might as well move this to the Ui folder.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**